controllers/azuremachine_reconciler.go

```diff
@@ -101,10 +102,13 @@ func newAzureMachineService(machineScope *scope.MachineScope) (*azureMachineServ
 			virtualmachinesSvc,
 			roleAssignmentsSvc,
 			vmextensionsSvc,
-			tagsSvc,
 		},
 		skuCache: cache,
 	}
+	if !strings.EqualFold(machineScope.CloudEnvironment(), azure.StackCloudName) {
+		ams.services = append(ams.services, tagsSvc)
+	}
```

Comment:
Is this needed because Azure Stack doesn't support the tags service? If so, I think a code comment here would be helpful.